### ISSUE TITLE
Deceptive Chapters on https://dragontea.ink/

### ISSUE THREAD

**Issue by aelementj** [dteviot/WebToEpub#687]
[https://dragontea.ink/](url) made the actual text on their chapters into garbled letters. On their website, it's exactly what is shown; but when i used WebToEpub on their chapter list, the text in their chapters became like this:

"**Cf Aruzm dlpv fk drgs z szmtlevi svzwzxsv. Srh hrtsg dzh uroovw drgs yvvi xzmh**"

Text shown on their website:
"_Zu Zifan woke up with a hangover headache. His sight was filled with beer cans..._"

Is it possible to negate this countermeasure and actually have chapters like shown on their website?

**Comment by dteviot** [comment]
@aelementj 
The bad news.  This is being done to stop people copying the site.  Modifying WebToEpub to bypass this is a violation of Google's Terms and Conditions and would get WebToEpub banned.

The good news: There is a tool to fix the epubs WebToEpub creates.
1. Go to https://drive.google.com/drive/folders/1B_X2WcsaI_eg9yA-5bHJb8VeTZGKExl8 and download EpubEditor.
2. Unzip the file and open main.html in Chrome or Firefox.
3. Replace the text in the box on the RIGHT with the following
```javascript
let decryptTable = new Map();
let crypt = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ";
let clear = "zyxwvutsrqponmlkjihgfedcbaZYXWVUTSRQPONMLKJIHGFEDCBA";
for(let i = 0; i < crypt.length; ++i) {
    decryptTable.set(crypt[i], clear[i]);
}
let decryptChar = c => decryptTable.get(c) ?? c;
let decryptString = cypherText => cypherText.split("").map(c => decryptChar(c)).join("");
for(let e of dom.querySelectorAll("p")) {
    e.textContent = decryptString(e.textContent);
}
return true;
```
4. Drop epub with scrambled text into box on LEFT.
5. Click option "Run script above to modify Epub"
6. Save result, when it finishes.

Tested with 1 chapter of https://dragontea.ink/novel/boss-level-face-slapping-specialist/

Time taken 23 minutes.

**Comment by aelementj** [comment]
Yoo~, thank you so much for the solution. I don't know much about coding so i literally just copied the "main.html" file to the computer and opened it on there. Little did i know that you need the rest of the contents of the folder in order for it to show the "Run script above to modify epub". I've learnt a lot on this exchange. I hope to acquire your help when another issue ensues.

tl;dr: tysm, I'll ask for help when another trouble comes up. Also, INT+1
